core/frontend/src/SubCategoriesCache.ts

```diff
@@ -74,6 +74,45 @@ export class SubCategoriesCache {
     };
   }
 
+  /**
+   * Populates the notLoadedCategoryIds property of the HydrateViewStateRequestProps.
+   * notLoadedCategoryIds is a subset of categoryIds, filtering out any ids which already have an entry in the cache.
+   */
+  public preload(options: HydrateViewStateRequestProps, categoryIds: Id64Arg): void {
+    let missing: Id64Set | undefined;
+    for (const catId of Id64.iterable(categoryIds)) {
+      if (undefined === this._byCategoryId.get(catId)) {
+        if (undefined === missing)
+          missing = new Set<string>();
+
+        missing.add(catId);
+      }
+    }
+
+    if (undefined === missing)
+      return;
+    options.notLoadedCategoryIds = CompressedId64Set.sortAndCompress(missing);
+  }
+
+  /**
+   * Populates the SubCategoriesCache using the categoryIdsResult of the HydrateViewStateResponseProps
+   */
+  public postload(options: HydrateViewStateResponseProps, categoryIds: Id64Arg): void {
+    // TODO: need missing here how wdo we get it without calculating again?
```

Comment:
Do you intend to address this TODO or the seemingly related question at the end of this function?

Reply:
Thanks for bringing that to my attention, I made the change